typescript/web/src/components/datasets/__tests__/delete-dataset-modal.tsx

```diff
@@ -1,99 +1,65 @@
+import { render, screen, fireEvent, waitFor } from "@testing-library/react";
+import { v4 as uuidV4 } from "uuid";
 import {
-  render,
-  screen,
-  fireEvent,
-  waitFor,
-  act,
-} from "@testing-library/react";
-import { ApolloProvider, gql } from "@apollo/client";
-import { PropsWithChildren } from "react";
-
-import { client } from "../../../connectors/apollo-client/schema-client";
-import { setupTestsWithLocalDatabase } from "../../../utils/setup-local-db-tests";
-import { DeleteDatasetModal } from "../delete-dataset-modal";
-
-const Wrapper = ({ children }: PropsWithChildren<{}>) => (
-  <ApolloProvider client={client}>{children}</ApolloProvider>
-);
+  getApolloMockWrapper,
+  ApolloMockedResponse,
+} from "../../../utils/testing/mock-apollo";
+import {
+  DeleteDatasetModal,
+  getDatasetByIdQuery,
+  deleteDatasetByIdMutation,
+} from "../delete-dataset-modal";
+import { mockDatasetSimple } from "../../../utils/testing/mock-data";
 
-setupTestsWithLocalDatabase();
+const mockQueries: Record<string, ApolloMockedResponse> = {
```

Comment:
I agree that using a "standardised" fixtures file for consistency is good, but this component doesn't use Apollo in its stories, so I didn't make a fixtures file for it, since it would only be used for `__tests__` yet.

What do you think should be done in this kind of case?

Reply:
Some goals of such harmonization:
* Reduce the amount of mental charge needed to understand tests made by others.
* Quickly get new tests up and running. By using "standardized" names, we can just scaffold new tests from existing ones without having to rename all of the variables and functions present in every tests.
* Encourage factorization. Since the blocks are already there, it just becomes easier to reuse them in other contexts.

In that sense, I'd say :
* At-least try to use the same `APOLLO_MOCKS` name in every tests so that we can start to have a reusable patterns.
* I'd be less strict with the `.fixtures` files I think...
  They are mostly useful to share data between tests, yes. Not only a single jest and storybook test couple, but also between multiple tests in the app.
  But besides that, it doesn't adds that much value so I guess we can just keep the correct names in the `.test` and `.stories` files as long as the code blocks are not too big (_which is what I don't really like with the Apollo mocks_) and don't need to be shared. We could still extract them in a `.fixtures` file.

This is the rule of thumb I've ended-up with for now, the `type TestCase`, `const TEST_CASES`, `function runTest` pattern you can see in all of my PR... But it still needs to be refined with feedbacks such as yours :)

I know it can seems a bit overthinking sometimes but at the end, it's a combination of those little things which makes us design new features faster while minimizing the impact on robustness.